src/clib/pio_nc4.c

```diff
@@ -84,11 +84,8 @@ PIOc_def_var_deflate(int ncid, int varid, int shuffle, int deflate,
     if (ios->ioproc)
     {
 #ifdef _NETCDF4
-        if (file->iotype == PIO_IOTYPE_NETCDF4P)
-            ierr = NC_EINVAL;
-        else
-            if (file->do_io)
-                ierr = nc_def_var_deflate(file->fh, varid, shuffle, deflate, deflate_level);
+        if (file->do_io)
+            ierr = nc_def_var_deflate(file->fh, varid, shuffle, deflate, deflate_level);
```

Comment:
I actually already have this fix, and a test for it, standing by to merge. I'm just getting the testing situation a bit more sorted out with GitHub actions, then I will merge the fix and test.

Reply:
OK - would you like me to pull the latest and remove this commit?

Reply:
No we can merge this if it passes all testing, then I will merge my test for it.

Reply:
This change got merged some time ago.